orchagent/flex_counter/flowcounterrouteorch.cpp

```diff
@@ -0,0 +1,985 @@
+#include "dbconnector.h"
+#include "directory.h"
+#include "flow_counter_handler.h"
+#include "logger.h"
+#include "routeorch.h"
+#include "flowcounterrouteorch.h"
+#include "schema.h"
+#include "swssnet.h"
+#include "table.h"
+#include "vnetorch.h"
+
+#include <string>
+
+extern Directory<Orch*>  gDirectory;
+extern RouteOrch*        gRouteOrch;
+extern size_t            gMaxBulkSize;
+extern sai_route_api_t*  sai_route_api;
+extern sai_object_id_t   gVirtualRouterId;
+extern sai_object_id_t   gSwitchId;
+
+#define FLEX_COUNTER_UPD_INTERVAL                   1
+#define FLOW_COUNTER_ROUTE_KEY                      "route"
+#define FLOW_COUNTER_SUPPORT_FIELD                  "support"
+#define ROUTE_PATTERN_MAX_MATCH_COUNT_FIELD         "max_match_count"
+#define ROUTE_PATTERN_DEFAULT_MAX_MATCH_COUNT       30
+#define ROUTE_FLOW_COUNTER_POLLING_INTERVAL_MS      10000
+
+FlowCounterRouteOrch::FlowCounterRouteOrch(swss::DBConnector *db, const std::vector<std::string> &tableNames):
+Orch(db, tableNames),
+mAsicDb(std::shared_ptr<DBConnector>(new DBConnector("ASIC_DB", 0))),
+mCounterDb(std::shared_ptr<DBConnector>(new DBConnector("COUNTERS_DB", 0))),
+mVidToRidTable(std::unique_ptr<Table>(new Table(mAsicDb.get(), "VIDTORID"))),
+mPrefixToCounterTable(std::unique_ptr<Table>(new Table(mCounterDb.get(), COUNTERS_ROUTE_NAME_MAP))),
+mPrefixToPatternTable(std::unique_ptr<Table>(new Table(mCounterDb.get(), COUNTERS_ROUTE_TO_PATTERN_MAP))),
+mRouteFlowCounterMgr(ROUTE_FLOW_COUNTER_FLEX_COUNTER_GROUP, StatsMode::READ, ROUTE_FLOW_COUNTER_POLLING_INTERVAL_MS, false),
+gRouteBulker(sai_route_api, gMaxBulkSize)
+{
+    SWSS_LOG_ENTER();
+    initRouteFlowCounterCapability();
+    if (mRouteFlowCounterSupported)
+    {
+        auto intervT = timespec { .tv_sec = FLEX_COUNTER_UPD_INTERVAL , .tv_nsec = 0 };
```

Comment:
I think this should be done if user enables the feature first time.

Reply:
We don't enable the timer here, we just create it, so it would not affect the performance if used does not enable route flow counter. We must create timer inside ctor or the timer will never run. Please check OrchDaemon::start(), it add all selectable at the begin of the function.
```cpp
void OrchDaemon::start()
{
    SWSS_LOG_ENTER();

    for (Orch *o : m_orchList)
    {
        m_select->addSelectables(o->getSelectables());
    }
```